trieste/space.py

```diff
@@ -339,3 +339,117 @@ def __mul__(self, other: Box) -> Box:
 
     def __deepcopy__(self, memo: dict[int, object]) -> Box:
         return self
+
+
+class OrdinalSearchSpace(Box):
+    r"""
+    Ordinal search space representing a :math:`D`-dimensional box in
```

Comment:
a *discrete* D-dimensional ...

Reply:
Say something like: Generates evenly-spaced values in an interval for each dimension